code/modules/mob/living/living_fov.dm

```diff
@@ -0,0 +1,262 @@
+		QDEL_NULL(blocker_mask)
+	if(visual_shadow) // In a case of early deletion due to volatile client
+		QDEL_NULL(visual_shadow)
+	return ..()
+
+/// Updates the state of the owner living and updates masking
+/datum/field_of_view/proc/update_living()
+	if(owner.stat == DEAD)
+		user_living = FALSE
+	else
+		user_living = TRUE
+	update_masking()
+
+/// Updates the state of the owner eye and updates masking
+/datum/field_of_view/proc/update_eye()
+	var/client/parent_client = owner.client
+	if(parent_client.eye == owner)
+		user_extends_eye = FALSE
+	else
+		user_extends_eye = TRUE
+	update_masking()
+
+/// Properly applies or removes the mask, on conditions of living and not extending an eye
+/datum/field_of_view/proc/update_masking()
+	var/should_apply_mask = FALSE
+	if(user_living && !user_extends_eye)
+		should_apply_mask = TRUE
+	if(should_apply_mask == applied_mask)
+		return
+	if(should_apply_mask)
+		add_mask()
+	else
+		remove_mask()
+
+/// Updates the size of the FOV masks by comparing them to client view size.
+/datum/field_of_view/proc/update_fov_size()
+	var/client/parent_client = owner.client
+	if(!parent_client) //Love client volatility!!
+		return
+	var/list/view_size = getviewsize(parent_client.view)
+	if(view_size[1] == current_fov_x && view_size[2] == current_fov_y)
+		return
+	current_fov_x = BASE_FOV_MASK_X_DIMENSION
+	current_fov_y = BASE_FOV_MASK_Y_DIMENSION
+	var/matrix/new_matrix = new
+	var/x_shift = view_size[1] - current_fov_x
+	var/y_shift = view_size[2] - current_fov_y
+	var/x_scale = view_size[1] / current_fov_x
+	var/y_scale = view_size[2] / current_fov_y
+	current_fov_x = view_size[1]
+	current_fov_y = view_size[2]
+	visual_shadow.transform = blocker_mask.transform = new_matrix.Scale(x_scale, y_scale)
+	visual_shadow.transform = blocker_mask.transform = new_matrix.Translate(x_shift * 16, y_shift * 16)
+
+/// Adds the masks to the user
+/datum/field_of_view/proc/add_mask()
+	var/client/parent_client = owner.client
+	if(!parent_client) //Love client volatility!!
+		return
+	parent_client.screen += blocker_mask
+	parent_client.screen += visual_shadow
+	applied_mask = TRUE
+
+/// Removes the masks from the user
+/datum/field_of_view/proc/remove_mask()
+	var/client/parent_client = owner.client
+	if(!parent_client) //Love client volatility!!
+		return
+	parent_client.screen -= blocker_mask
+	parent_client.screen -= visual_shadow
+	applied_mask = FALSE
+
+/// When a direction of the user changes, so do the masks
+/datum/field_of_view/proc/dir_change(new_dir)
+	blocker_mask.dir = new_dir
+	visual_shadow.dir = new_dir
+
+/datum/field_of_view/proc/set_fov_angle(new_angle)
+	fov_angle = new_angle
+	blocker_mask.icon_state = "[fov_angle]"
+	visual_shadow.icon_state = "[fov_angle]_v"
+
+/// Is `observed_atom` in a mob's field of view? This takes blindness, nearsightness and FOV into consideration
+/mob/living/proc/in_fov(atom/observed_atom, ignore_self = FALSE)
```

Comment:
why is this a proc on the mob and not on the datum (taking a source atom).

Reply:
In previous reply I clarified that the datum is the client handler. You may want to check `in_fov()` whether that mob has a client with a fov handler or not. Currently this being on mob/living isn't nessecary as we only use it to play the effects which already require a client, but for any expansions upon the system I imagine this will be useful.